File: packages/shared_preferences/shared_preferences_android/android/src/main/java/io/flutter/plugins/sharedpreferences/SharedPreferencesListEncoder.java
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package io.flutter.plugins.sharedpreferences;

import androidx.annotation.NonNull;
import java.util.List;

/** SharedPreferencesPlugin */
public interface SharedPreferencesListEncoder {
  @NonNull
  String encode(@NonNull List<String> list);

  @NonNull
  List<String> decode(@NonNull String listString);
}

File: packages/shared_preferences/shared_preferences_android/android/src/main/kotlin/io/flutter/plugins/sharedpreferences/SharedPreferencesListEncoder.kt
// Copyright 2013 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
package io.flutter.plugins.sharedpreferences

/**
 * An interface used to provide conversion logic between List<String> and String for
 * SharedPreferencesPlugin.
 */
interface SharedPreferencesListEncoder {
  /** Converts list to String for storing in shared preferences. */
  fun encode(list: List<String>): String

  /** Converts stored String representing List<String> to List. */
  fun decode(listString: String): List<String>
}
